### ISSUE TITLE
Can beam_decode_trans use torch.tensor to run on gpu

### ISSUE THREAD

**Issue by wjhuang0113** [pooyafayyaz/Fingerspelling-PoseNet#6]
Hi,

I discovered that the reason for the long evaluation time when running train_with_lm.py might be due to line 165:

current_preds = decoder_dec.beam_decode_trans(log_probs[0].detach().cpu(), beam_size, model, poses, beta=lm_beta, gamma=ins_gamma)
I suspect that log_probs[0] needs to be converted to a NumPy array before it can be used in beam_decode_trans, and this might be causing the slowdown.

Additionally, I’d like to ask how I could speed up this part. Theoretically, inputting a torch.tensor directly into beam_decode_trans could significantly speed things up, but I’m not sure how to make this modification here. Could you provide some suggestions? Thank you!

**Comment by pooyafayyaz** [comment]
Hi,

I don't believe converting the NumPy array directly to a tensor will have a significant impact. The process could be sped up by batching the candidates and processing them together, but that falls outside the scope of this work. However, if you manage to create a version that implements this, I'd be happy to merge it into the main repository.
